circuit-retained/src/commonMain/kotlin/com/slack/circuit/retained/RememberRetained.kt

```diff
@@ -88,81 +65,162 @@ public fun <T : Any> rememberRetained(vararg inputs: Any?, key: String? = null,
       compositeKey.toString(MaxSupportedRadix)
     }
 
+  @Suppress("UNCHECKED_CAST") (saver as Saver<T, Any>)
+
   val canRetainChecker = LocalCanRetainChecker.current ?: rememberCanRetainChecker()
   val holder =
     remember(canRetainChecker) {
-      // value is restored using the registry or created via [init] lambda
-      val restored = registry.consumeValue(finalKey) as? RetainableHolder.Value<*>
-      val finalValue = restored?.value ?: init()
-      val finalInputs = restored?.inputs ?: inputs
-      RetainableHolder(
-        registry = registry,
+      // value is restored using the retained registry first, the saveable registry second, or
+      // created via [init]
+      // lambda third
+      @Suppress("UNCHECKED_CAST")
+      val retainedRestored =
+        retainedStateRegistry.consumeValue(finalKey) as? RetainableSaveableHolder.Value<T>
+      val saveableRestored =
+        saveableStateRegistry?.consumeRestored(finalKey)?.let { saver.restore(it) }
+      val finalValue = retainedRestored?.value ?: saveableRestored ?: init()
+      val finalInputs = retainedRestored?.inputs ?: inputs
+      RetainableSaveableHolder(
+        retainedStateRegistry = retainedStateRegistry,
         canRetainChecker = canRetainChecker,
+        saveableStateRegistry = saveableStateRegistry,
+        saver = saver,
         key = finalKey,
         value = finalValue,
         inputs = finalInputs,
-        hasBeenRestored = restored != null,
+        hasBeenRestoredFromRetained = retainedRestored != null,
       )
     }
   val value = holder.getValueIfInputsAreEqual(inputs) ?: init()
-  SideEffect { holder.update(registry, finalKey, value, inputs) }
-  @Suppress("UNCHECKED_CAST") return value as T
+  SideEffect {
+    holder.update(retainedStateRegistry, saveableStateRegistry, saver, finalKey, value, inputs)
+  }
+  return value
 }
 
+/**
+ * Remember the value produced by [init].
+ *
+ * It behaves similarly to [remember], but the stored value will survive configuration changes, such
+ * as a screen rotation.
+ *
+ * Use this overload if you remember a mutable state.
```

Comment:
Updated - this matches the overloads of `rememberSaveable`, where it can be convenient to just write the `Saver` for the thing inside of `mutableStateOf`, instead of the whole `MutableState`